nms/README.md

```diff
@@ -1,14 +1,141 @@
+Specifically, triggering unit tests, e2e tests, eslint, and flow tests are done from this directory.
+
+## Running Local Dev NMS
+> NOTE: This guide is written for development directly using Docker.
+> You may need to make adjustments if using MiniKube.
+
+### Pre-requisites
+The NMS requires a working connection to the Magma Orc8r to function.
+For local development, bring up the Orc8r docker containers.
+See `magma/orc8r/cloud/docker/docker-compose.yml`
+To ensure that NMS has access to metrics and graphing functionality, bring up the Prometheus and Grafana containers.
+See `magma/orc8r/cloud/docker/docker-compose.metrics.yml`
+
+### Build NMS Docker Image
+Run the following command from `magma/nms/packages/magmalte`
+```
+docker-compose build magmalte
+```
+
+### Run NMS Containers
+Run the following command from `magma/nms/packages/magmalte`
+```
+docker-compose up -d
+```
+
+## Checking NMS Logs
+Run the following from `magma/nms/packages/magmalte`
+```
+docker-compose logs -f magmalte
+```
+Depending on the issues you run into, you may need to check the logs of other containers, both NMS and Orc8r related.
+
+## Testing
+
+### Install Dependencies
+Running eslint and flow tests requires installing dependencies.
+Run the following command from `magma/nms` to install these dependencies:
+```
+yarn
+```
+
+### Eslint
+Run from `magma/nms`
+```
+yarn run eslint ./
+```
+
+### Flow
+Run from `magma/nms`
+```
+flow
+```
+
+### Unit Tests
+Run from `magma/nms`
+```
+yarn run test
+```
+
+### Testing Coverage
 
 Run `yarn test --coverage`
+
+## Updating NMS for API Changes
+
+### How to re-generate MagmaAPIBindings.js
+1. Place an up-to-date `swagger.yml` in `magma/nms/packages/magmalte`
+   You can get the full `swagger.yml` at `{orc8r domain}/swagger/v1/spec`
+   (e.g. `https://localhost:9443/swagger/v1/spec`)
+2. Run `scripts/generateAPIFromSwagger.sh` from `magma/nms/packages/magmalte`
+
+
+## Accessing Local Dev NMS
+
+### Multitenancy and Organizations
+
+Multitenancy is supported in the Magma NMS. Each tenant is called an "organization".
+Each organization owns a subset of the networks provisioned on Orchestrator, and the special `master` organization administrates organizations in the system.
+
+Users in organizations log into the NMS using a subdomain that matches their organization name.
+For example, users of a *NewOrg* organization in the NMS would access the NMS using http://NewOrg.localhost:8081/nms
+
+Note that any NMS user can only access the organization it was created uner.
+
+### First-time Setup
+When you deploy the NMS for the first time, you'll need to create a user that has access to the master organization.
+
+Run the following command from `magma/nms/packages/magmalte` and make sure to substitute `ADMIN_USER_EMAIL` and `ADMIN_USER_PASSWORD` with your desired email and password.
+```
+docker-compose exec magmalte yarn setAdminPassword master ADMIN_USER_EMAIL ADMIN_USER_PASSWORD
```

Comment:
side note: i noticed while doing this that password set here isn't enforced by validation req of >10 chars, vs passwords created in the UI are validated. Are we doing client side validation for password requirements?

Reply:
I'll add that in a separate fix, though hopefully we can get away entirely from using these sorts of scripts.